packages/bpk-component-calendar/src/BpkColoredCalendarDate.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2020 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { cssModules } from 'bpk-react-utils';
+
+import BpkCalendarDate, { propTypes, defaultProps } from './BpkCalendarDate';
+import STYLES from './BpkColoredCalendarDate.scss';
+
+const getClassName = cssModules(STYLES);
+
+export const CellType = {
+  default: 'default',
+  negative: 'negative',
+  neutral: 'neutral',
+  positive: 'positive',
+};
+
+type Props = {
+  cellType: String,
+  isFocused: boolean,
+  isSelected: boolean,
+};
+
+const BpkColoredCalendarDate = (props: Props) => {
+  const { cellType, isFocused, isSelected, ...rest } = props;
+  const className = getClassName(
+    'bpk-colored-calendar-date',
+    cellType === CellType.default && 'bpk-colored-calendar-date--default',
+    cellType === CellType.positive && 'bpk-colored-calendar-date--positive',
+    cellType === CellType.neutral && 'bpk-colored-calendar-date--neutral',
+    cellType === CellType.negative && 'bpk-colored-calendar-date--negative',
+    isFocused && 'bpk-colored-calendar-date--focused',
+    isSelected && 'bpk-colored-calendar-date--selected',
+  );
+
+  return <BpkCalendarDate {...rest} className={className} />;
+};
+
+BpkColoredCalendarDate.propTypes = {
+  ...propTypes,
+  cellType: PropTypes.oneOf(Object.keys(CellType)),
+};
+BpkColoredCalendarDate.defaultProps = {
+  ...defaultProps,
+  cellType: null,
```

Comment:
Shouldn't `default` be the default prop here? If we want to have null then the flow type needs to be `?$Keys<typeof CellType>`

Reply:
I will make it null-able as the `default` cellType has a different meaning, so might also make sense to rename it to avoid confusion